lib/auth/app.ex

```diff
@@ -0,0 +1,142 @@
+defmodule Auth.App do
+  @moduledoc """
+  Schema and helper functions for creating/managing Apps.
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+  import Ecto.Query, warn: false
+  alias Auth.Repo
+  # https://stackoverflow.com/a/47501059/1148249
+  alias __MODULE__
+
+  schema "apps" do
+    field :desc, :binary
+    field :end, :naive_datetime
+    field :name, :binary
+    field :url, :binary
+    field :person_id, :id
+    field :status, :id
+    has_many :apikeys, Auth.Apikey
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(app, attrs) do
+    app
+    |> cast(attrs, [:name, :desc, :url, :end, :person_id, :status])
+    |> validate_required([:name, :url])
+  end
+
+  @doc """
+  Returns the list of apps.
+
+  ## Examples
+
+      iex> list_apps()
+      [%App{}, ...]
+
+  """
+  def list_apps do
+    App
+    |> where([a], a.status != 6)
+    |> Repo.all()
+  end
+
+  @doc """
+  Gets a single app.
+
+  Raises `Ecto.NoResultsError` if the App does not exist.
+
+  ## Examples
+
+      iex> get_app!(123)
+      %App{}
+
+      iex> get_app!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  def get_app!(id) do
+    App
+    |> where([a], a.id == ^id and a.status != 6)
+    |> Repo.one()
+    |> Repo.preload(:apikeys)
+  end
+
+  @doc """
+  Creates a app.
+
+  ## Examples
+
+      iex> create_app(%{field: value})
+      {:ok, %App{}}
+
+      iex> create_app(%{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def create_app(attrs \\ %{}) do
+    case %App{} |> App.changeset(attrs) |> Repo.insert() do
+      {:ok, app} ->
+        # Create API Key for App https://github.com/dwyl/auth/issues/97
+        %{"app" => app, "status" => 3}
+        |> AuthWeb.ApikeyController.make_apikey(app.person_id)
```

Comment:
Calling a controller function here feels wrong - `make_apikey/1` should be in `Auth.App` or `Auth.ApiKey` contexts

Reply:
@th0mas totally agree. This is a legacy from ages ago when I was first creating the API Keys. Very happy for this to be refactored. (_I'm busy writing some code that touches this file so maybe hold off on refactoring till all tests are passing again ..._)

Reply:
`AuthWeb.ApikeyController.make_apikey/1` removed completely. ✅